src/utils/basic_multicall.cairo

```diff
@@ -1,7 +1,28 @@
+use satoru::utils::error::UtilsError;
+use starknet::{account::Call, call_contract_syscall, SyscallResultTrait};
+
 /// Receives and executes a batch of function calls on this contract.
 /// # Arguments
 /// * `data` - calls to execute.
-fn multicall(data: Array<Array<felt252>>) -> Array<Array<felt252>> { // TODO
-    ArrayTrait::new()
+fn multicall(mut data: Array<Call>) -> Array<Span<felt252>> {
+    assert(data.len() > 0, UtilsError::NO_DATA_FOR_MULTICALL);
+    let mut result = ArrayTrait::new();
+    loop {
+        match data.pop_front() {
+            Option::Some(call) => {
+                let mut res = call_contract_syscall(
+                    address: call.to,
```

Comment:
Call should be made to address of current contract

Reply:
Thx for the comment.
I wanted to keep it flexible. No worries i will replace call.to by get_contract_address().
Do you have a use case in mind so i can build a test case accordingly ?

Reply:
Discussed with Titan. Let’s keep the flexible solution

Reply:
works for me !